test/tests/performance/pbs_sched_perf.py

```diff
@@ -216,3 +216,33 @@ def test_pset_fuzzy_perf(self):
             cycle1_time, cycle2_time, (cycle1_time / cycle2_time) * 100))
         self.assertLess(cycle2_time, cycle1_time,
                         'Optimization was not faster')
+
+    @timeout(3600)
+    def test_sched_query(self):
+        """
+        Test scheduler's turnaround time with focus on querying the universe
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+
+        # We have 10010 vnodes, submit 10050 jobs so that it can run almost
+        # all of them
+        a = {'Resource_List.select': '1:ncpus=1'}
+        jids = self.submit_jobs(attribs=a, num=10050, step=0, wt_start=900)
+
+        # Kick a sched cycle which will run 1010 jobs
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+        jid1010 = jids[1009]
+        self.server.expect(JOB, {"job_state": "R"}, jid1010)
```

Comment:
Use a job_state=R count instead of checking for a specific job.  You're making an assumption that the scheduler chose all the jobs before that.  It should, but you never know what might change in the future.